tas/forms.py

```diff
@@ -158,15 +158,16 @@ class UserProfileForm(forms.Form):
     citizenshipId = forms.CharField(label='Country of citizenship', widget=forms.HiddenInput(), error_messages={'invalid': 'Please select your Country of citizenship'})
 
     def __init__(self, *args, **kwargs):
+        is_federated = kwargs.pop('is_federated', False)
         super(UserProfileForm, self).__init__(*args, **kwargs)
         self.fields['institutionId'].choices = get_institution_choices()
-
         data = self.data or self.initial
         if (data is not None and 'institutionId' in data and data['institutionId']):
             self.fields['departmentId'].choices = get_department_choices(data['institutionId'])
-
+        if is_federated:
+            for field in self.fields:
+                self.fields[field].widget.attrs['readonly'] = True
```

Comment:
Will this disable the PI eligibility request checkbox? As discussed, that's the only bit we need to remain interactive. I am not sure how else the PI eligibility can be requested, unless I have forgotten a detail.

Reply:
No, Request PI Eligibility is not a property on the form by default, instead the ability to make a PI eligibility request (checkbox) is added manually in the template and handled as a special case on POST
https://github.com/ChameleonCloud/portal/blob/pi_eligibility/tas/templates/tas/profile_edit.html#L18
https://github.com/ChameleonCloud/portal/pull/94/files#diff-edcdfb0669e5546028c9cb2a21085ec1L69

Reply:
OK, thanks for the clarification!